lib/instrumentation/modules/http2.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+var debug = require('debug')('elastic-apm')
+var eos = require('end-of-stream')
+var shimmer = require('../shimmer')
+
+module.exports = function (http2, agent) {
+  debug('shimming http2.createServer function')
+  shimmer.wrap(http2, 'createServer', wrapCreateServer)
+  shimmer.wrap(http2, 'createSecureServer', wrapCreateServer)
+
+  return http2
+
+  // The `createServer` function will unpatch itself after patching
+  // the first server prototype it patches.
+  function wrapCreateServer (original) {
+    return function wrappedCreateServer (options, handler) {
+      var server = original.apply(this, arguments)
+      shimmer.wrap(server.constructor.prototype, 'emit', wrapEmit)
+      wrappedCreateServer.__obUnwrap()
```

Comment:
What's the reason for un-patching this?

Reply:
(Fun fact: `ob` is short for `Opbeat` - We should probably rename these at some point)

Reply:
It only needs to wrap the emit method once. I could also have just used a Boolean check, but it's marginally better performance if I just restore the original function after the first run.

Reply:
Ah true. Missed that